worker/upgradeseries/worker.go

```diff
@@ -268,6 +268,18 @@ func (w *upgradeSeriesWorker) handleCompleteStarted() error {
 	return nil
 }
 
+// handleCompleted givens the worker a chance to perform any final operations
+// before returning control to the server (controller) which will then perform its own
+// post upgrade routine.
+func (w *upgradeSeriesWorker) handleCompleted() error {
```

Comment:
Nothing calls this at the moment.

The way the other `handle{Thing}` members are laid out, is that they are invoked based on the machine status **only**. Inside the handlers there are calls the API to check unit statuses.

So I imagine this would be called from another case and look something like:

```go
	units, allConfirmed, err = w.compareUnitAgentServices(w.UnitsCompleted)
	if err != nil {
		return errors.Trace(err)
	}
	if allConfirmed {
		w.logger.Infof("series upgrade complete")
		return errors.Trace(w.FinishUpgradeSeries())
	}
```

Reply:
Yes, we don't need to check the unit status, because we know machine could not be complete without that.